src/lightning/pytorch/profilers/pytorch.py

```diff
@@ -314,6 +317,19 @@ def __init__(
                 f"Found sort_by_key: {self._sort_by_key}. Should be within {self.AVAILABLE_SORT_KEYS}. "
             )
 
+        for key in self._table_kwargs:
+            if key in {"sort_by", "row_limit"}:
+                raise KeyError(
+                    f"Found invalid table_kwargs key: {key}. This is already a positional argument of the Profiler."
+                )
+            valid_table_keys = set(inspect.signature(EventList.table).parameters.keys()) - {
+                "self",
+                "sort_by",
+                "row_limit",
+            }
+            if key not in valid_table_keys:
+                raise KeyError(f"Found invalid table_kwargs key: {key}. Should be within {valid_table_keys}.")
```

Comment:
This validation is a bit overkill. I don't think we do this anywhere else we pass kwargs

Reply:
Yes in retrospect this probably isn't necessary. The reason I did this though is to make it fail immediately as opposed to when the table summary is created.

Shall I change it to this structure? I.e. just let it run with valid kwargs.
```
    def _create_profiler(self, profiler: Type[_PROFILER]) -> _PROFILER:
        init_parameters = inspect.signature(profiler.__init__).parameters
        kwargs = {k: v for k, v in self._profiler_kwargs.items() if k in init_parameters}
        return profiler(**kwargs)
```

Reply:
It's okay. Let's leave it as it is.

The profiler logic is because there's different profiler classes